src/daft-connect/src/convert.rs

```diff
@@ -0,0 +1,45 @@
+mod data_conversion;
+mod expression;
+mod formatting;
+mod plan_conversion;
+mod schema_conversion;
+
+use std::{
+    collections::HashMap,
+    ops::{ControlFlow, Try},
+    sync::Arc,
+};
+
+use common_daft_config::DaftExecutionConfig;
+use daft_plan::LogicalPlanRef;
+use daft_table::Table;
+pub use data_conversion::convert_data;
+use eyre::Context;
+pub use plan_conversion::to_logical_plan;
+pub use schema_conversion::connect_schema;
+
+pub fn run_local<T: Try>(
+    logical_plan: &LogicalPlanRef,
+    mut f: impl FnMut(&Table) -> T,
+    default: impl FnOnce() -> T,
+) -> eyre::Result<T> {
+    let physical_plan = daft_physical_plan::translate(logical_plan)?;
+    let cfg = Arc::new(DaftExecutionConfig::default());
+    let psets = HashMap::new();
+
+    let stream = daft_local_execution::run_local(&physical_plan, psets, cfg, None)
+        .wrap_err("running local execution")?;
+
+    for elem in stream {
+        let elem = elem?;
+        let tables = elem.get_tables()?;
+
+        for table in tables.as_slice() {
+            if let ControlFlow::Break(x) = f(table).branch() {
+                return Ok(T::from_residual(x));
+            }
+        }
+    }
+
+    Ok(default())
+}
```

Comment:
Why can't we just return a stream of `Table` instead of this callback pattern? This seems overly complex to me.

Reply:
Yeah, I agree the pattern being used right now is not ideal. I think it's being used because I actually tried to implement a stream pattern, but there are issues. In particular, you're consuming values as they're created, rather than calling `.next` on a stream to gather values. It seemed like there might have been a need for cloning or something. I'm not really sure, but I'm going to re-examine it because I agree it's quite complex and quite ugly.

Reply:
i think if we created an async version of [run_local](https://github.com/universalmind303/Daft/blob/afefdcef4f87df10cbb1cbb5ef6ce195bd8ffaf1/src/daft-local-execution/src/run.rs#L114), it would solve a lot of the problems you're running into. 

It would make it natively a stream instead of converting the iter into one and possible collecting intermediate values. 

It would also remove the issue of spawning a runtime inside of a runtime.